File: removeMoreThan2.80.ts
function removeDuplicates(nums: Number[]) {
  let elem = 0;
  let counter: { [key: number]: number } = {};
  let removeIndexes = [];
  for (let i = 0; i < nums.length; i++) {
    counter[nums[i]] = 0;
  }
  for (let j = 0; j < nums.length; j++) {
    if (counter[nums[j]] < 2) {
      counter[nums[j]]++;
      elem++;
    } else {
      removeIndexes.push(j);
    }
  }

  nums = nums.filter((_, i) => !removeIndexes.includes(i));
  console.log(removeIndexes);
  console.log(nums, elem);
  return elem;
}

removeDuplicates([1, 2, 3, 4, 5, 5, 5, 2, 2]);

File: removeMoreThan2.80.py
def removeDuplicates(self, nums: List[int]) -> int:
    count = 0
    x = {x: 0 for x in nums}
    removeIndexes = []
    for i in range(len(nums)):
        if x[nums[i]] < 2:
            x[nums[i]] += 1
            count += 1
        else:
            removeIndexes.append(i)

    nums[:] = [nums[i] for i in range(len(nums)) if i not in removeIndexes]
    return count
